libsolidity/formal/ModelChecker.cpp

```diff
@@ -17,6 +17,9 @@
 // SPDX-License-Identifier: GPL-3.0
 
 #include <libsolidity/formal/ModelChecker.h>
+#ifdef HAVE_Z3_DLOPEN
+#include <libsmtutil/Z3Interface.h>
+#endif
```

Comment:
Not sure I get this change. Why only in that case?

Reply:
Only then we need it for ``Z3Interface::available()``.
We could also use ``#ifdef HAVE_Z3`` here and always use ``Z3Interface::available()`` in ``availableSolvers()`` below, though, since that's constant ``true``, unless ``HAVE_Z3_DLOPEN`` anyways...

Reply:
Ah right, yes. I think the suggestion would simplify it a bit

Reply:
Ok, force-push adjusted it in the last commit.